src/Config_Command.php

```diff
@@ -127,10 +127,22 @@ public function create( $_, $assoc_args ) {
 			$assoc_args['extra-php'] = file_get_contents( 'php://stdin' );
 		}
 
-		// TODO: adapt more resilient code from wp-admin/setup-config.php
 		if ( ! \WP_CLI\Utils\get_flag_value( $assoc_args, 'skip-salts' ) ) {
-			$assoc_args['keys-and-salts'] = self::_read(
-				'https://api.wordpress.org/secret-key/1.1/salt/' );
+			try {
+				$assoc_args['keys-and-salts'] = true;
+				$assoc_args['auth-key'] = self::unique_key();
+				$assoc_args['secure-auth-key'] = self::unique_key();
+				$assoc_args['logged-in-key'] = self::unique_key();
+				$assoc_args['nonce-key'] = self::unique_key();
+				$assoc_args['auth-salt'] = self::unique_key();
+				$assoc_args['secure-auth-salt'] = self::unique_key();
+				$assoc_args['logged-in-salt'] = self::unique_key();
+				$assoc_args['nonce-salt'] = self::unique_key();
+			} catch ( Exception $e ) {
```

Comment:
@danielbachhuber Oh duh - my bad. I (wrongly) assumed the try/catch block [used in core](https://github.com/WordPress/WordPress/blob/dd46042e3c3b43c5ee3c17862213b0280fb5abfd/wp-admin/setup-config.php#L286-L296) was to catch whether the `random_int()` was available, but you're right, obviously that would throw a uncatchable fatal!

I see now it's to catch something completely different:

> If an appropriate source of randomness cannot be found, an Exception will be thrown.

So the try/catch is still needed, but you're right that a `function_exists( 'random_int' )` should be here too.